ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -889,18 +897,19 @@ def set_waypoints_velocity(self):
                                 lookahead_wps)
         else:
             if self.waypoints[self.final_waypoints_start_ptr].get_v() <\
-                    self.handoff_velocity:
+                    min(self.handoff_velocity, self.waypoints[self.final_waypoints_start_ptr].get_maxV()):
```

Comment:
Actually, I found a bug where the 'maintainspeed' state breaks when the current velocity hits the max velocity.  Line 891 above needs the same change as this line 900 where the check needs to become `min(self.handoff_velocity, self.waypoints[self.final_waypoints_start_ptr].get_maxV())`.